avm/res/network/express-route-port/main.bicep

```
@@ -0,0 +1,207 @@
+
+@description('The resource group of the ExpressRoute Gateway was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The name of the ExpressRoute Gateway.')
+output name string = expressRoutePort.name
+
+@description('The location the resource was deployed into.')
+output location string = expressRoutePort.location
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type roleAssignmentType = {
+  @description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
+  name: string?
+
+  @description('Required. The role to assign. You can provide either the display name of the role definition, the role definition GUID, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+  roleDefinitionIdOrName: string
+
+  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
+  principalId: string
+
+  @description('Optional. The principal type of the assigned principal ID.')
+  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?
+
+  @description('Optional. The description of the role assignment.')
+  description: string?
+
+  @description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container".')
+  condition: string?
+
+  @description('Optional. Version of the condition.')
+  conditionVersion: '2.0'?
+
+  @description('Optional. The Resource Id of the delegated managed identity resource.')
+  delegatedManagedIdentityResourceId: string?
+}[]?
+
+type linkType = {
+  @description('Optional. Resource Id of the existing Link.')
+  id: 'string'?
+
+  @description('Required. The name of the link to be created.')
+  name: 'string'
+
+  @description('Optional. Properties of the Link.')
+  properties: {
+    @description('Required. Administrative state of the physical port.')
+    adminState: 'string'
+
+    @description('Optional. MacSec Configuration of the link.')
+    macSecConfig: {
+      @description('Required. Keyvault Secret Identifier URL containing Mac security CAK key.')
+      cakSecretIdentifier: 'string'
+
+      @description('Required. Mac security cipher.')
+      cipher: ('GcmAes128' | 'GcmAes256' | 'GcmAesXpn128' | 'GcmAesXpn256')
+
+      @description('Required. Keyvault Secret Identifier URL containing Mac security CKN key.')
+      cknSecretIdentifier: 'string'
+
+      @description('Required. Sci mode.')
+      sciState: ('Enabled' | 'Disabled')
+    }?
+  }?
+}[]?
```

```suggestion
}
```
Should be moved to parameter that implements the type as per https://azure.github.io/Azure-Verified-Modules/specs/bcp/res/#id-bcpnfr18---user-defined-types---specification